x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -195,6 +195,7 @@ public void putEngine(Engine engine, ActionListener<IndexResponse> listener) {
                     )
                     .endObject();
             }
+            // TODO Validate indices exist?
```

Comment:
I was thinking on validating that the indices exist here. WDYT?

Reply:
Would it be better to put that validation in the `TransportPutEngineAction`?

It feels to me that the API layer is the place we want to do this validation and contract enforcement, not in the service layer. This way we can also ensure that the API caller has permissions on the individual indices if so desired. WDYT?